vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.html

```diff
@@ -24,309 +27,339 @@
-    $('div[data-link-id]').removeClass("alert-warning");
-    $('div[data-link-id]').addClass("alert-primary");
-    
-    // reset image links
-    $('img[data-link-id]').css("border-width", "0"); 
+function drawLinks(){
+    $("#main-space span").remove();
+	$("#main-space i").remove();
+	$("#space span").remove();
+	$("#space i").remove();
+    $.ajax({
+		type: "GET",
+		 url: "[(@{'/staff/space/'+${space.id}+'/links?'+${_csrf.parameterName}+'='+${_csrf.token}})]",
+		cache       : false,
+        contentType : false,
+        processData : false,
+        success: function(data) {
+        	var spaceLinks = data['spaceLinks'];
+        	var externalLinks = data['externalLinks'];
+        	var moduleLinks = data['moduleLinks'];
+        	var posXModal = $("#bgImage").position().left;
+        	var posYModal = $("#bgImage").position().top;
+        	var posX = $("#main-bgImage").position().left;
+        	var posY = $("#main-bgImage").position().top;
+        	var img = document.getElementById("bgImage");
+    		var imageHeight = img.height;
+            var imageWidth = img.width;
+            var displayWidth = [(${display.width})];
+            var displayHeight = [[${display.height}]];
+            function setLinkPosition(modalLink, link, isSpaceLink, isModal, isExternalLink){
+                modalLink.css('position', 'absolute');
+                if(isModal){
+                    var	linkPosX=(imageWidth*link.positionX)/displayWidth;
+                    var	linkPosY=(imageHeight*link.positionY)/displayHeight;
+                    modalLink.css('left',linkPosX + posXModal + 20);
+                    modalLink.css('top', linkPosY + posYModal + 20);
+                }else{
+                    modalLink.css('left', link.positionX + posX);
+                    modalLink.css('top', link.positionY + posY);
+                }
+                modalLink.css('transform', 'rotate('+link.rotation+'deg)');
+                modalLink.find("span").css('fill', 'grey');
+                modalLink.css('color', 'rgba(128,128,128,1)');
+                modalLink.css('font-size', "12px");
+                setInputElements(link,isModal,isExternalLink);
+                if(isSpaceLink && link.link.targetSpace.spaceStatus=="UNPUBLISHED"){
+                    var unpublishedSpaceElement;
+                    unpublishedSpaceElement=$('<i data-link-id="unpublished-'+link.link.id+'" class="fa fa-exclamation-triangle fa-lg unpublishedSpaceClass unpublishedClass-'+link.link.id+'" style="color: var(--secondary);"></i>')
+        			unpublishedSpaceElement.css('position', 'absolute');
+        		    if(isModal){
+        		        unpublishedSpaceElement.css('left', linkPosX + posXModal + 25);
+        		        unpublishedSpaceElement.css('top', linkPosY + posYModal - 13);
+        		    }else{
+        		        unpublishedSpaceElement.css('left', link.positionX + posX + 25);
+            		    unpublishedSpaceElement.css('top', link.positionY + posY - 13);

```

Comment:
same as above, what are these magic numbers for?

Reply:
to adjust the position of an unpublished link icon on the head of actual link

Reply:
if I ask a question like that, it usually means it needs some commenting ;) In this case, they should also defined somewhere at the beginning as constants so it's easy to adjust if necessary).